### ISSUE TITLE
Grey Roofs when using Levels and Wall Height modules

### ISSUE THREAD

**Issue by Larkinabout** [dev7355608/vision-5e#66]
When using Vision 5e with the Levels and Wall Height modules, roofs appear grey when a token with vision is selected. If either Vision 5e  or Levels/Wall Height is disabled, the roofs gain colour again. The roofs are visible in either instance.

**Versions**
Foundry 11.315
dnd5e 2.4.1
Vision 5e  1.12.0
Levels 4.3.1
Wall Height 5.1.4

With Vision 5e, Levels and Wall Height enabled:
![image](https://github.com/dev7355608/vision-5e/assets/105953297/19db9fdc-ee92-4e12-a906-01f867adf591)

With Vision 5e disabled:
![image](https://github.com/dev7355608/vision-5e/assets/105953297/904da0f3-9014-47da-872f-38095c792dab)

**Comment by dev7355608** [comment]
This is probably just the way the Darkvision vision mode behaves. I assume Vision 5e selects Darkvision for this token automatically. You should be able to reproduce it without Vision 5e, I think: compare Basic Vision vs. Darkvision.

Though I don't remember why I didn't make the Vision 5e' Darkvision mode adaptive such that the desaturation is linked to the scene's darkness level. The Darkvision mode of core isn't adaptive. Having it be adaptive seems to make more sense assuming Darkness Level is used for the time of day. I'll have to think about it again.

**Comment by Larkinabout** [comment]
I fear I missed off an important factor and led you wrongly.

In the scene, light was being provided by a light source with 'Height (Top)' and 'Height (Bottom)' set to 'Infinity'. This was with the intention of mimicking daylight outside but darkness inside; though I've since learned that roofs do not currently block light, so this doesn't actually do what I'd intended. 

As part of testing your theory of darkvision, I've now enabled 'Global Illumination' and the roofs are showing in colour with Vision 5e enabled. Darkvision on tokens does not appear to impact this, though the token previously selected did not have darkvision.

I'm not sure why a light with no top and bottom limit causes grey roofs, that's still an issue, but until roofs block light, it's probably not an issue that will come up very often.
